### PR TITLE
DAOS-623 actions: Improve github actions support.

### PR DESCRIPTION
**Pull request by ashleypittman**
Add another build test to github-actions.  This doesn't
modify the existing code, but intends to replace it once
stable.

Use the github cache to re-use docker images for speed.
Do a single image create, then re-use this for different
compilers etc.
Run NLT in docker, and report server leaks as annotations.
Remove PSM2 from the Docker builds.

This is enabled to build centos.8 with gcc/clang, and to
build and test with ubuntu/clang but can be readily
extended to the rest of the matrix once landed and stable
without adding additional time or complexity.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by jolivier23** [comment]
Do we not still need to keep opa?  Have we replaced all of the legacy hardware?

File: utils/sl/components/__init__.py [hunk 1]
```diff
@@ -66,13 +66,6 @@ def check(self, name):
         self.not_installed.append(name)
         return False
 
-def exclude(reqs, name, use_value, exclude_value):
```
Comment (jolivier23) [inline utils/sl/components/__init__.py:69]: If we remove this, do we also need to remove the EXCLUDE Variable?   Presently, it is only used for psm2 exclusion

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Adding request changes to get it off my inbox until question(s) are addressed.

File: utils/sl/components/__init__.py (hunk 1 above)
Comment (ashleypittman) [reply utils/sl/components/__init__.py:69]: Possibly, you mean as a command line option?  I can't rule out needing this function again at some point so I was on the fence about leaving this in.

**Comment by ashleypittman** [comment]
> Do we not still need to keep opa? Have we replaced all of the legacy hardware?

I didn't change anything with openpa did I?  I've no idea if we still need it or not, but it's outside the scope of this change.

File: utils/sl/components/__init__.py (hunk 1 above)
Comment (jolivier23) [reply utils/sl/components/__init__.py:69]: yes, this implements the command line option by checking if psm2 is included on the command line.

**Comment by jolivier23** [comment]
> > Do we not still need to keep opa? Have we replaced all of the legacy hardware?
> 
> I didn't change anything with openpa did I? I've no idea if we still need it or not, but it's outside the scope of this change.

I mean opa the fabric, not openpa the atomic library.  Actually opa2 which is what psm2 is for.   I think a lot of our internal nodes may still have these cards but I'm not sure.  @johannlombardi ?

**Review by frostedcmos — DISMISSED** [review]

File: utils/sl/components/__init__.py (hunk 1 above)
Comment (ashleypittman) [reply utils/sl/components/__init__.py:69]: This bit isn't actually required for the rest of the functionality, I only included it as Johann had said that he thought we'd already remove it.

I did try testing with a different distro at one point, and psm2 doesn't compile there which is why it got removed, but that's no longer part of this PR.

**Comment by ashleypittman** [comment]
> > > Do we not still need to keep opa? Have we replaced all of the legacy hardware?
> > 
> > 
> > I didn't change anything with openpa did I? I've no idea if we still need it or not, but it's outside the scope of this change.
> 
> I mean opa the fabric, not openpa the atomic library. Actually opa2 which is what psm2 is for. I think a lot of our internal nodes may still have these cards but I'm not sure. @johannlombardi ?

I've removed that part of the PR, it was incidental to the rest of the work.

**Review by jolivier23 — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**